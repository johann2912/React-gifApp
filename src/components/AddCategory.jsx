import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('Batman')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    };

    const onSubmits = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim();
        if(newInputValue.length <= 1) return;

        onNewCategory(newInputValue);
        setInputValue('');
    };

    return (
        <form onSubmit={onSubmits}>
            <input 
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}