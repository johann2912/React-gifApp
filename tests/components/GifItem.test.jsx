import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests in <GifItem />', () => {

    const title = 'Vegeta';
    const url   = 'https://dbz-vegeta.com/vegeta.jpg';

    test('should match with snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />)
        expect( container ).toMatchSnapshot();
    })

    test('Should image with URL and ALT', () => {
        render(<GifItem title={ title } url={ url } />);

        expect( screen.getByRole('img').src).toBe(url);
        expect( screen.getByRole('img').alt).toBe(title);
    })

    test('Should title in an component', () => {
        render(<GifItem title={ title } url={ url } />);

        expect( screen.getByText(title)).toBeTruthy();        
    })

})