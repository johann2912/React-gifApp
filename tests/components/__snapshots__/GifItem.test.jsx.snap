// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Vegeta"
      src="https://dbz-vegeta.com/vegeta.jpg"
    />
    <p>
      Vegeta
    </p>
  </div>
</div>
`;
